### ISSUE TITLE
Implement commonly used decoders (`syslog`, `json`)

### ISSUE THREAD

**Issue by pablosichert** [vectordotdev/vector#7465]
Depends on #7463.

At least the following decoders should be available for all sources:

- `json`
- `syslog`

**Comment by jszwedko** [comment]
👍 I think we also want `syslog` here too.

**Comment by pablosichert** [comment]
Could that be an alias for `octet`? I'll need to review the details, but at least that's where the inspiration for the `octet` framing comes from.

**Comment by jszwedko** [comment]
I think so. I think the framing and encoding are separate concerns. For example, an octet framed RFC 5424 message:

```
107 <34>1 2019-07-11T22:14:15.003Z aliyun.example.com ali - ID47 - BOM'su root' failed for lonvick on /dev/pts/8
```

The `107 ` bit would be the framing.

But then we also need to be able to decode

```
<34>1 2019-07-11T22:14:15.003Z aliyun.example.com ali - ID47 - BOM'su root' failed for lonvick on /dev/pts/8
```

As a syslog encoded message.

This would facilitate, for example, decoding syslog messages from Kafka events or other sources as well as let us implement the `syslog` source as simply a `socket` source with a `syslog` codec.

**Comment by pablosichert** [comment]
Ah, yeah that makes sense! Thanks for the example, added it to the list.

**Comment by jeromekleinen-kbc** [comment]
I'm not sure how feasible this is, but imagine you could configure some VRL statements directly in the Sources (and by extension in the Sinks as well). Personally, I would use this to call parse_json directly in my Sources where I currently use a small transform. Perhaps writing VRL is a bit more daunting to novice users as compared to setting ```encoding = "json"``` or something, but in terms of flexibility, code reuse and consistency this seems hard to beat :)

**Comment by pablosichert** [comment]
@jeromekleinen-kbc interesting suggestion! We could very well add a feature along the lines of

```
encoding.type = "remap"
encoding.source = "..."
```

in the future.

**Comment by pablosichert** [comment]
Closed with #9055, #9065 and #9066.
